src/codecs/decoding/decoder.rs

```diff
@@ -138,4 +138,33 @@ mod tests {
         let event = next.unwrap().0.pop().unwrap().into_log();
         assert_eq!(event.get("bar").unwrap(), &Value::from(2));
     }
+
+    #[tokio::test]
+    async fn gelf_streaming_decoding() {
```

Comment:
Added this test, I don't know if its correct to place it here. Didn't find anywhere else to place it.

I'm new in this project and can't find some modules easily. 
Maybe this test is more of an integration test, but couldn't find any related in the root `/tests` directory. I'm missing something?

It would be nice to have something like a TCP socket integration test where we test this same behavior...As the `default_stream_framing` is also called in https://github.com/vectordotdev/vector/blob/e982f6679e9d1526d856efa462e728774b52cf34/src/sources/socket/mod.rs#L123.

I thought of adding a similar test like this in https://github.com/vectordotdev/vector/blob/e982f6679e9d1526d856efa462e728774b52cf34/src/sources/socket/mod.rs#L391, what do you think?
In that case, I would create a TCP config setting only `decoding = DeserializerConfig::from(GelfDeserializerConfig)`(and leaving the framing as None in that config) and see how the TCP initialization takes the default framer.

What are your thoughts of all of this?